packages/ui/react-ui-editor/src/components/Toolbar/Toolbar.tsx

```diff
@@ -137,6 +140,91 @@ const ToolbarButton = ({ Icon, children, ...props }: ToolbarButtonProps) => {
   );
 };
 
+//
+// View Mode
+//
+
+const ViewModes: { [key: string]: { label: string; icon: Icon } } = {
```

Comment:
from the screen shot: make sure icons and labels are in a grid.
and try to favor label not breaking.